exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporterBuilder.java

```diff
@@ -125,33 +126,76 @@ public HttpExporterBuilder<T> exportAsJson() {
+   * </ul>
+   */
+  private HttpSender createHttpSender() {
     Map<String, String> headers = this.headers == null ? Collections.emptyMap() : this.headers;
     Supplier<Map<String, String>> headerSupplier = () -> headers;
 
-    HttpSender httpSender = null;
-    // TODO: once we publish multiple HttpSenderProviders, log warning when multiple are found
-    for (HttpSenderProvider httpSenderProvider :
+    HttpSenderProvider httpSenderProvider;
+    String configuredSender =
+        ConfigUtil.getString("io.opentelemetry.exporter.internal.http.HttpSenderProvider", "");
+    Map<String, HttpSenderProvider> httpSenderProviders = new HashMap<>();
+    for (HttpSenderProvider spi :
         ServiceLoader.load(HttpSenderProvider.class, HttpExporterBuilder.class.getClassLoader())) {
-      httpSender =
-          httpSenderProvider.createSender(
-              endpoint,
-              compressionEnabled,
-              exportAsJson ? "application/json" : "application/x-protobuf",
-              timeoutNanos,
-              headerSupplier,
-              authenticator,
-              retryPolicy,
-              tlsConfigHelper.getSslContext(),
-              tlsConfigHelper.getTrustManager());
-      LOGGER.log(Level.FINE, "Using HttpSender: " + httpSender.getClass().getName());
-      break;
+      httpSenderProviders.put(spi.getClass().getName(), spi);
     }
-    if (httpSender == null) {
+    if (httpSenderProviders.isEmpty()) {
       throw new IllegalStateException(
-          "No HttpSenderProvider found on classpath. Please add dependency on opentelemetry-exporter-http-sender-okhttp");
+          "No HttpSenderProvider found on classpath. Please add dependency on "
+              + "opentelemetry-exporter-http-sender-okhttp or opentelemetry-exporter-http-sender-jdk");
+    } else if (httpSenderProviders.size() == 1) {
+      httpSenderProvider = httpSenderProviders.values().stream().findFirst().get();
+    } else { // Multiple HttpSenderProviders
+      if (configuredSender.isEmpty()) { // Multiple providers but none configured
+        LOGGER.log(
+            Level.WARNING,
+            "Multiple HttpSenderProvider found. Please include only one, "
+                + "or specify preference setting io.opentelemetry.exporter.internal.http.HttpSenderProvider "
+                + "to the FQCN of the preferred provider.");
+        httpSenderProvider = httpSenderProviders.values().stream().findFirst().get();
+      } else if (httpSenderProviders.containsKey(
+          configuredSender)) { // Multiple providers, configured matches
+        httpSenderProvider = httpSenderProviders.get(configuredSender);
+      } else { // Multiple providers, configured does not match
+        throw new IllegalStateException(
+            "No HttpSenderProvider matched configured io.opentelemetry.exporter.internal.http.HttpSenderProvider: "
+                + configuredSender);
+      }
```

Comment:
seems like all of this work to figure out the provider probably belongs in a method dedicated to that purpose. I think it'll be a clearer implementation if it can return early when it found a resolved sender, rather than having all of the if/else branches.

Reply:
Much better.